.travis.yml

```diff
@@ -11,6 +11,9 @@ go:
 install:
   - make bootstrap
 before_script:
+  # Download protoc3.
+  - curl -OL https://github.com/google/protobuf/releases/download/v3.2.0/protoc-3.2.0-linux-x86_64.zip && unzip protoc-3.2.0-linux-x86_64.zip -d protoc3 && sudo mv protoc3/bin/* /usr/local/bin/ && sudo mv protoc3/include/* /usr/local/include/ 
+
```

Comment:
Can you elaborate as to why you are switching to proto3? There are some; let's say; interesting design choices made as to how it handles empty values for example.

Reply:
yes @AmitKumarDas , that would be good. i will do that. will add pb.go file also in to revision control

Reply:
done